### PR TITLE
Consolidate conditional compilation

### LINKED ISSUE #45758: Consider hiding nullability differences in PEModuleBuilder.Translate()
A couple of recent PRs have changed `MetadataWriter` to accommodate type references that differ only by nullability: https://github.com/dotnet/roslyn/pull/38249 and https://github.com/dotnet/roslyn/pull/45714.

Since nullability is not persisted as part of the type reference, an alternative suggested by @AlekseyTs to changing `MetadataWriter` might be to hide nullability differences in `PEModuleBuilder.Translate()`.

For instance, the following change to `PEModuleBuilder.Translate()` seems to work as an alternative to https://github.com/dotnet/roslyn/pull/45714.

```C#
internal Cci.INamedTypeReference Translate(
    NamedTypeSymbol namedTypeSymbol,
    SyntaxNode syntaxNodeOpt,
    DiagnosticBag diagnostics,
    bool fromImplements = false,
    bool needDeclaration = false)
{
    Debug.Assert(namedTypeSymbol.IsDefinitionOrDistinct());
    Debug.Assert(diagnostics != null);

    // If the only differences between the type and OriginalDefinition is nullability, use
    // OriginalDefinition because nullability is not persisted as part of the type reference.
    if (!namedTypeSymbol.IsDefinition)
    {
        var definition = namedTypeSymbol.OriginalDefinition;
        if (namedTypeSymbol.Equals(definition, TypeCompareKind.AllNullableIgnoreOptions))
        {
            namedTypeSymbol = definition;
        }
    }

    // ...
```

See also related issue https://github.com/dotnet/roslyn/pull/40271 translating [native integers](https://github.com/dotnet/roslyn/commit/f22ad6e4b43b0c0495989e74d6f764cb456727dc#diff-3e769ab816ce69b4ac1b52b10107fb26).

### PR DESCRIPTION
**Pull request by AlekseyTs**
This is a follow up on two previous PRs.
Closes #45758.

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Emitter/Model/NamespaceSymbolAdapter.cs [hunk 1]
```diff
@@ -16,17 +16,10 @@ internal partial class
 #endif 
         Cci.INamespace
     {
-#if DEBUG
-        internal NamespaceSymbolAdapter(NamespaceSymbol underlyingNamespaceSymbol)
-        {
-            AdaptedNamespaceSymbol = underlyingNamespaceSymbol;
-        }
+        Cci.INamespace Cci.INamespace.ContainingNamespace => AdaptedNamespaceSymbol.ContainingNamespace?.GetCciAdapter();
+        string Cci.INamedEntity.Name => AdaptedNamespaceSymbol.MetadataName;
```
Comment (333fred) [inline src/Compilers/CSharp/Portable/Emitter/Model/NamespaceSymbolAdapter.cs:20]: Nit: please either put a newline between the above properties, or remove the one below.

**Review by 333fred — APPROVED** [review]
Thanks for consolidating. This LGTM, one minor formatting nit.

**Comment by AlekseyTs** [comment]
@cston, @dotnet/roslyn-compiler Please review, need a second sign-off.

**Comment by cston** [comment]
Are the deletes and adds of code blocks in the Adapter files just code moves?

**Comment by AlekseyTs** [comment]
> Are the deletes and adds of code blocks in the Adapter files just code moves?

Yes. Just moved members around

**Review by cston — APPROVED** [review]

**Pull request merged**